.github/workflows/test.yml

```diff
@@ -49,3 +49,29 @@ jobs:
       - name: 'Tear down Docker'
         if: always()
         run: docker stop --time 0 mir-semantics-ci-${GITHUB_SHA}
+
+  nix:
+    name: 'Nix'
+    needs: code-quality-checks
+    runs-on: ubuntu-latest
```

Comment:
For this PR we are just going to focus on this architecture. We will make another issue and PR to extend to other OS